stan/math/prim/prob/chi_square_lccdf.hpp

```diff
@@ -51,59 +51,58 @@ return_type_t<T_y, T_dof> chi_square_lccdf(const T_y& y, const T_dof& nu) {
 
   scalar_seq_view<T_y> y_vec(y);
   scalar_seq_view<T_dof> nu_vec(nu);
+  size_t size_nu = stan::math::size(nu);
   size_t N = max_size(y, nu);
 
   operands_and_partials<T_y, T_dof> ops_partials(y, nu);
 
   // Explicit return for extreme values
   // The gradients are technically ill-defined, but treated as zero
   for (size_t i = 0; i < stan::math::size(y); i++) {
-    if (value_of(y_vec[i]) == 0) {
+    const T_partials_return y_dbl = value_of(y_vec[i]);
+    if (y_dbl == 0) {
       return ops_partials.build(0.0);
     }
+    if (y_dbl == INFTY) {
+      return ops_partials.build(NEGATIVE_INFTY);
+    }
   }
 
   using std::exp;
   using std::log;
   using std::pow;
 
+  VectorBuilder<true, T_partials_return, T_dof> half_nu(size_nu);
+  VectorBuilder<!is_constant_all<T_y, T_dof>::value, T_partials_return, T_dof>
+      tgamma_vec(size_nu);
   VectorBuilder<!is_constant_all<T_dof>::value, T_partials_return, T_dof>
-      gamma_vec(size(nu));
-  VectorBuilder<!is_constant_all<T_dof>::value, T_partials_return, T_dof>
-      digamma_vec(size(nu));
-
-  if (!is_constant_all<T_dof>::value) {
```

Comment:
Was there a bug here? It feels like the logic changed from checking only T_dof to T_dof and T_y

Reply:
For `digamma_vec` nothing has changed, only the diff is a bit hard to read.

What changed is the logic for `tgamma_vec`: I've done this so that this precomputed term can be used further down in a `if (!is_constant_all<T_y>::value)` block, where before we were recomputing that term.